src/palimpzest/query/optimizer/rules.py

```diff
@@ -256,27 +254,24 @@ def substitute(cls, logical_expression: LogicalExpression, **physical_op_params)
         return deduped_physical_expressions
 
 
-class LLMConvertRule(ImplementationRule):
+class LLMConvertBondedRule(ImplementationRule):
     """
-    Base rule for bonded and conventional LLM convert operators; the physical convert class
-    (LLMConvertBonded or LLMConvertConventional) is provided by sub-class rules.
-
-    NOTE: we provide the physical convert class(es) in their own sub-classed rules to make
-    it easier to allow/disallow groups of rules at the Optimizer level.
+    Substitute a logical expression for a ConvertScan with a bonded convert physical implementation.
     """
 
-    # overridden by sub-classes
-    physical_convert_class = None
-
     @classmethod
     def matches_pattern(cls, logical_expression: LogicalExpression) -> bool:
-        is_match = isinstance(logical_expression.operator, ConvertScan) and logical_expression.operator.udf is None
-        logger.debug(f"LLMConvertRule matches_pattern: {is_match} for {logical_expression}")
+        is_match = (
+            isinstance(logical_expression.operator, ConvertScan) and logical_expression.operator.udf is None
+            # Use bonded convert only if there are multiple generated fields
+            # and len(logical_expression.operator.get_generated_fields()) > 1
```

Comment:
can we delete this commented code now? or do you want to keep it around for the future?

Reply:
oops, my bad.